backend/server.js

```diff
@@ -1,106 +1,236 @@
+      return res.status(400).json({ message: 'All fields are required' });
     }
+
+    // Create user
     const hashedPassword = await bcrypt.hash(password, 10);
-    const user = new User({ firstName, lastName, email, phone, password: hashedPassword });
+    const user = new User({
+      firstName,
+      lastName,
+      email,
+      phone,
+      password: hashedPassword
+    });
+
     await user.save();
-    res.status(201).send('User registered successfully');
-  } catch (err) {
-    console.error('Error registering user:', err);
-    res.status(500).send('Internal server error');
+
+    // Generate token
+    const token = jwt.sign({ userId: user._id }, JWT_SECRET, { expiresIn: '1h' });
+
+    res.status(201).json({
+      message: 'Registration successful',
+      token,
+      user: {
+        id: user._id,
+        firstName: user.firstName,
+        lastName: user.lastName,
+        email: user.email,
+        role: user.role
+      }
+    });
+  } catch (error) {
+    next(error);
   }
 });
 
-// User login
-app.post('/login', async (req, res) => {
-  const { email, password } = req.body;
-  if (!email || !password) {
-    return res.status(400).send('All fields are required');
-  }
+app.post('/api/auth/login', async (req, res, next) => {
   try {
+    const { email, password } = req.body;
+
+    if (!email || !password) {
+      return res.status(400).json({ message: 'Email and password are required' });
+    }
+
     const user = await User.findOne({ email });
-    if (!user || !(await bcrypt.compare(password, user.password))) {
-      return res.status(400).send('Invalid credentials');
+    if (!user) {
+      return res.status(401).json({ message: 'Invalid credentials' });
+    }
+
+    const isValidPassword = await bcrypt.compare(password, user.password);
+    if (!isValidPassword) {
+      return res.status(401).json({ message: 'Invalid credentials' });
+    }
+
+    const token = jwt.sign({ userId: user._id }, JWT_SECRET, { expiresIn: '1h' });
+
+    res.json({
+      message: 'Login successful',
+      token,
+      user: {
+        id: user._id,
+        firstName: user.firstName,
+        lastName: user.lastName,
+        email: user.email,
+        role: user.role
+      }
+    });
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Appointment routes
+app.post('/api/appointments', authenticateToken, async (req, res, next) => {
+  try {
+    const { fullName, email, phone, department, appointmentDate, appointmentTime, symptoms } = req.body;
+
+    // Validate required fields
+    if (!fullName || !email || !phone || !department || !appointmentDate || !appointmentTime) {
+      return res.status(400).json({ message: 'All required fields must be filled' });
     }
-    const token = jwt.sign({ userId: user._id }, '123', { expiresIn: '1h' });
-    res.send({ token, message: 'Login successful' });
-  } catch (err) {
-    console.error('Error during login:', err);
-    res.status(500).send('Internal server error');
+
```

Comment:
Using the JWT token with appointment api request for creating the appointments for verified users